app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id', counter_cache: :posts_counter
+  has_many :likes
+  has_many :comments
+
+  def update_posts_counter
+    author.increment!(:posts_counter)
+  end
```

Comment:
- [ **OPTIONAL** ] For best practices, I think it would be nice if you use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method `private`. i.e.
```ruby
class Post < ApplicationRecord
  ...

  def recent_comments
    comments.order(created_at: :desc).limit(5)
  end

  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
end
```

Reply:
@tresorsawasawa Thank. would adjust

Reply:
resolved :+1: